openslides_backend/action/actions/user/create.py

```diff
@@ -47,8 +50,37 @@ class UserCreate(
     )
 
     def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        if not (
+            instance.get("username")
+            or instance.get("first_name")
+            or instance.get("last_name")
+        ):
+            raise ActionException("Need username or first_name or last_name")
+        if not instance.get("username"):
+            instance["username"] = self.generate_username(instance)
         if not instance.get("default_password"):
             instance = self.generate_and_set_password(instance)
         else:
             instance = self.set_password(instance)
         return super().update_instance(instance)
+
+    def generate_username(self, instance: Dict[str, Any]) -> str:
+        count = 0
+
+        while True:
+            new_username = instance.get("first_name", "") + instance.get(
```

Comment:
Please add a space there, if both parts are given.

Reply:
Add a space if both are set. If first_name is "Jon" and last_name "Smith" and it is the third one, the username is "Jon Smith2".